Userland/Libraries/LibWeb/HTML/Parser/HTMLTokenizer.swift

```diff
@@ -0,0 +1,138 @@
+        case ScriptDataEscapeStartDash
+        case ScriptDataEscaped
+        case ScriptDataEscapedDash
+        case ScriptDataEscapedDashDash
+        case ScriptDataEscapedLessThanSign
+        case ScriptDataEscapedEndTagOpen
+        case ScriptDataEscapedEndTagName
+        case ScriptDataDoubleEscapeStart
+        case ScriptDataDoubleEscaped
+        case ScriptDataDoubleEscapedDash
+        case ScriptDataDoubleEscapedDashDash
+        case ScriptDataDoubleEscapedLessThanSign
+        case ScriptDataDoubleEscapeEnd
+        case BeforeAttributeName
+        case AttributeName
+        case AfterAttributeName
+        case BeforeAttributeValue
+        case AttributeValueDoubleQuoted
+        case AttributeValueSingleQuoted
+        case AttributeValueUnquoted
+        case AfterAttributeValueQuoted
+        case SelfClosingStartTag
+        case BogusComment
+        case MarkupDeclarationOpen
+        case CommentStart
+        case CommentStartDash
+        case Comment
+        case CommentLessThanSign
+        case CommentLessThanSignBang
+        case CommentLessThanSignBangDash
+        case CommentLessThanSignBangDashDash
+        case CommentEndDash
+        case CommentEnd
+        case CommentEndBang
+        case DOCTYPE
+        case BeforeDOCTYPEName
+        case DOCTYPEName
+        case AfterDOCTYPEName
+        case AfterDOCTYPEPublicKeyword
+        case BeforeDOCTYPEPublicIdentifier
+        case DOCTYPEPublicIdentifierDoubleQuoted
+        case DOCTYPEPublicIdentifierSingleQuoted
+        case AfterDOCTYPEPublicIdentifier
+        case BetweenDOCTYPEPublicAndSystemIdentifiers
+        case AfterDOCTYPESystemKeyword
+        case BeforeDOCTYPESystemIdentifier
+        case DOCTYPESystemIdentifierDoubleQuoted
+        case DOCTYPESystemIdentifierSingleQuoted
+        case AfterDOCTYPESystemIdentifier
+        case BogusDOCTYPE
+        case CDATASection
+        case CDATASectionBracket
+        case CDATASectionEnd
+        case CharacterReference
+        case NamedCharacterReference
+        case AmbiguousAmpersand
+        case NumericCharacterReference
+        case HexadecimalCharacterReferenceStart
+        case DecimalCharacterReferenceStart
+        case HexadecimalCharacterReference
+        case DecimalCharacterReference
+        case NumericCharacterReferenceEnd
+    }
+
+    enum StopAtInsertionPoint {
+        case Yes
+        case No
+    }
+
+    var input = Swift.String()
+    var state = State.Data
+    var returnState = State.Data
+
+    var currentToken = HTMLToken()
+    var queuedTokens = Deque<HTMLToken>()
+
+    public init() {}
+    public init?(input: AK.StringView, encoding: AK.StringView) {
+        if let string = Swift.String(decoding: input, as: encoding) {
+            self.input = string
+        } else {
+            return nil
+        }
+    }
+
+    public func nextToken(stopAtInsertionPoint: StopAtInsertionPoint = StopAtInsertionPoint.No) -> HTMLToken? {
```

Comment:
Do we still want to use enums for bools in swift? Having function parameters already named makes it feel a little redundant.